src/string.cr

```diff
@@ -839,18 +869,31 @@ class String
     end
   end
 
+  # Returns a new String with *char* removed if the string ends with it.
+  #
+  # ```
+  # "hello".chomp('o') # => "hell"
+  # "hello".chomp('a') # => "hello"
+  # ```
   def chomp(char : Char)
     if ends_with?(char)
-      count = 0
-      char.each_byte do |byte|
-        count += 1
-      end
-      String.new(unsafe_byte_slice(0, bytesize - count))
+      String.new(unsafe_byte_slice(0, bytesize - char.bytesize))
     else
       self
     end
   end
 
+  # Returns a new String with *string* removed if the string ends with it.
```

Comment:
This seems kind of confusing to me: with string removed if the string ends with it. However, I can't think of anything better... :/